### PR TITLE
enum - TrackObject flags

### PR DESCRIPTION
**Pull request by niceeffort**
Converting enum flags to new style per #1757

### CODE REVIEW

File: src/OpenLoco/src/Objects/TrackObject.h [hunk 1]
```diff
@@ -72,6 +77,16 @@ namespace OpenLoco
         bool validate() const;
         void load(const LoadedObjectHandle& handle, stdx::span<const std::byte> data, ObjectManager::DependentObjects* dependencies);
         void unload();
+
+        constexpr bool hasFlags(Flags22 flagsToTest) const
+        {
+            return (flags & flagsToTest) != Flags22::none;
+        }
+
+        constexpr bool hasFlags(TrackPieceFlags flagsToTest) const
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Objects/TrackObject.h:86]: Same as road. Prefer not overloading

File: src/OpenLoco/src/Objects/TrackObject.h [hunk 2]
```diff
@@ -15,35 +16,39 @@ namespace OpenLoco
         struct RenderTarget;
     }
 
-    namespace Flags22
+    enum class Flags22 : uint16_t
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Objects/TrackObject.h:19]: Rename to track object flags. And add object to the piece flags

File: src/OpenLoco/src/Windows/Construction/ConstructionTab.cpp [hunk 3]
```diff
@@ -621,14 +621,10 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
     // 0x0049DAF3
     static void disableTrackSlopes(Window* self, TrackObject trackObj, uint64_t disabledWidgets)
     {
-        auto maskedTrackPieces = trackObj.trackPieces & ((1 << 5) | (1 << 8));
-
-        if (maskedTrackPieces != ((1 << 5) | (1 << 8)))
+        if (!trackObj.hasFlags(TrackPieceFlags::slope | TrackPieceFlags::slopedCurve))
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Windows/Construction/ConstructionTab.cpp:624]: This logic isn't quite the same. It should be two has flags calls && together. It's saying if both not (slope and curved slope). You've written if not ( slope or curved slope )

File: src/OpenLoco/src/Windows/Construction/ConstructionTab.cpp [hunk 4]
```diff
@@ -621,14 +621,10 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
...
             disabledWidgets |= (1 << widx::slope_down) | (1 << widx::slope_up);
 
-        maskedTrackPieces = trackObj.trackPieces & ((1 << 6) | (1 << 8));
-
-        if (maskedTrackPieces != ((1 << 6) | (1 << 8)))
+        if (!trackObj.hasFlags(TrackPieceFlags::steepSlope | TrackPieceFlags::slopedCurve))
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Windows/Construction/ConstructionTab.cpp:627]: Same here should be two has flags

Comment (niceeffort) [reply src/OpenLoco/src/Windows/Construction/ConstructionTab.cpp]: OK, I think I have it fixed on the latest checkin.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**